src/lib/containers/EntityBadgeIcons.tsx

```diff
@@ -0,0 +1,309 @@
+      } else if (annotationsCount) {
+        // If annotations exist, then indicate that they are invalid
+        setSchemaConformance(SchemaConformanceState.INVALID)
+      } else {
+        // If annotations are missing, it's still invalid, but should appear differently
+        setSchemaConformance(SchemaConformanceState.ANNOTATIONS_MISSING)
+      }
+    }
+  }, [
+    schemaValidationResults,
+    isInExperimentalMode,
+    annotationsCount,
+    bundle,
+    SchemaConformanceState.VALID,
+    SchemaConformanceState.INVALID,
+    SchemaConformanceState.ANNOTATIONS_MISSING,
+  ])
+
+  /**
+   * Convert the list of annotations to a string of <tr>...anno1</tr><tr>...anno2</tr>...
+   * If there are no annotations, this is the empty string ('')
+   */
+  const annotationsTableRows = bundle
+    ? `${Object.entries(bundle.annotations?.annotations ?? []).reduce(
+        (previous, current, index) => {
+          if (
+            index < maxAnnosToShow ||
+            (index === maxAnnosToShow && maxAnnosToShow === annotationsCount)
+          ) {
+            return `${previous}<tr><td><b>${
+              current[0]
+            }</b></td><td>${current[1].value.join(', ')}</td></tr>`
+          } else {
+            return previous
+          }
+        },
+        '',
+      )}`
+    : ''
+
+  // We also show the schema name if there is one (and we're in experimental mode)
+  const valiationSchemaTableRow =
+    isInExperimentalMode && boundSchema
+      ? `<tr><td><b>Validation Schema</b></td><td>${boundSchema.jsonSchemaVersionInfo.schemaName}</td></tr>`
+      : ''
+
+  // Format it all as an html table
+  const annotationsHtml = `
+      ${
+        schemaValidationResults ? `<p>${schemaConformance} Annotations</p>` : ''
+      }
+      <table>
+      ${annotationsTableRows ? annotationsTableRows : ''}
+      ${valiationSchemaTableRow}
+      </table>${
+        annotationsCount > maxAnnosToShow
+          ? `<p>and ${annotationsCount - maxAnnosToShow} more</p>`
+          : ''
+      }`
+
+  if (bundle == null) {
+    return null
+  }
+
+  return (
+    <div className="EntityBadge" style={{ flexWrap, justifyContent }}>
+      <ReactTooltip
+        id={ENTITY_BADGE_ICONS_TOOLTIP_ID}
+        className="EntityBadgeTooltip"
+        delayShow={100}
+        place={'right'}
+      />
+      <div onClick={e => e.stopPropagation()}>
+        <EntityModal
+          entityId={entityId}
+          show={showModal}
+          showTabs={false}
+          onClose={() => setShowModal(false)}
+          initialTab={EntityModalTabs.ANNOTATIONS}
+        />
+      </div>
+      {showIsPublicPrivate && bundle.benefactorAcl && isPublic(bundle) ? (
+        <FontAwesomeIcon
+          aria-hidden={false}
+          role="img"
+          className="EntityBadge__Badge"
+          icon={faGlobeAmericas}
+          data-for={ENTITY_BADGE_ICONS_TOOLTIP_ID}
+          data-tip={'Public'}
+          data-testid={'is-public-icon'}
+        />
+      ) : null}
```

Comment:
Why do you use the conditional ternary operator necessary here?  Is the 'null' value important?  (Why not just isPublic(bundle) && <FontAwesomeIcon...)

Reply:
Doesn't matter here, but trying to use ternaries more so as to not accidentally use `&&` [when it does matter](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx).